rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala

```diff
@@ -111,25 +112,17 @@ class ReportingRspace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P,
     * Seq(Seq[ReportingEvent](Precharge), Seq[ReportingEvent](userDeploy), Seq[ReportingEvent](Refund))
     * It would be seperated by the softcheckpoint creation.
     */
-  val report: SyncVar[Seq[Seq[ReportingEvent]]] = create[Seq[Seq[ReportingEvent]]](Seq.empty)
-  val softReport: SyncVar[Seq[ReportingEvent]]  = create[Seq[ReportingEvent]](Seq.empty)
+  val report: Ref[F, Seq[Seq[ReportingEvent]]] = Ref.unsafe(Seq.empty[Seq[ReportingEvent]])
+  val softReport: Ref[F, Seq[ReportingEvent]]  = Ref.unsafe(Seq.empty[ReportingEvent])
 
-  private def collectReport =
-    for {
-      sReport <- getSoftReport
-      _ = if (sReport.nonEmpty) {
-        report.update(s => s :+ sReport)
-        softReport.update(_ => Seq.empty[ReportingEvent])
-      }
-    } yield ()
+  private def collectReport: F[Unit] =
+    softReport.get.flatMap(
+      sReport =>
+        (report.update(s => s :+ sReport) >> softReport.update(_ => Seq())).whenA(sReport.nonEmpty)
+    )
```

Comment:
Curly brace can be useful to keep argument of anonymous function in the same line.
Prefer `*>` over `>>` because it requires only Applicative instead of Monad which is one tiny nesting less.
```scala
softReport.get.flatMap { sReport =>
  (report.update(_ :+ sReport) *> softReport.set(Seq())).whenA(sReport.nonEmpty)
}
```